### PR TITLE
Modernize step-24.

### PR DESCRIPTION
**Pull request by drwells**
This PR introduces several small changes:

1. Remove trailing whitespace.
2. Remove unnecessary default constructor definitions.
3. Combine declarations and definitions of some simple functions.
4. Remove some unused headers.
5. Clean up the discussion of boundaries.
6. Switch to VTU output.

### CODE REVIEW

File: examples/step-24/step-24.cc [hunk 1]
```diff
@@ -267,7 +257,7 @@ namespace Step24
   {
     const Point<dim> center;
     GridGenerator::hyper_ball(triangulation, center, 1.);
-    triangulation.refine_global(7);
+    triangulation.refine_global(8);
```
Comment (masterleinad) [inline examples/step-24/step-24.cc]: Why do you change the number of refinements? Is this consistent with the output?

Comment (drwells) [reply examples/step-24/step-24.cc]: No, this was a mistake: I changed it to check something but didn't mean to merge it. I will fix this.

**Comment by drwells** [comment]
Fixed.

**Pull request merged**